### DISCUSSION TITLE
Equivalent to PyTorch's GRU or LSTM in Lava-dl?

### DISCUSSION THREAD

**Discussion by ZoltanV-V** [lava-nc/lava-dl#180] [Q&A] [answered]
Hello, I am interested in developing a machine translation model using Lava and Loihi. I've done a couple of PyTorch implementations where I used nn.GRU and nn.LSTM. Is there something similar to these layers in either Slayer or Bootstrap? If not, are there available references that I can look at that might be able to help me understand how to create an equivalent model? Thank you so much!

**Comment by bamsumit** [comment, marked as answer]
Hi @ZoltanV-V, there is no direct equivalent for them in lava-dl. The closest thing available is the [recurrent block](https://lava-nc.org/lava-lib-dl/slayer/block/block.html#lava.lib.dl.slayer.block.cuba.Recurrent). However **there are all the primitives available in lava-dl to implement something like [LSNN](https://proceedings.neurips.cc/paper_files/paper/2018/file/c203d8a151612acf12457e4d67635a95-Paper.pdf)**

**Reply by ZoltanV-V** [reply]
Thank you so much, I'll take a look at the referenced paper and the recurrent block in Slayer also!
